docs/developer-guide/nodejs-api.md

```diff
@@ -427,7 +437,16 @@ The return value is an object containing the results of the linting operation. H
                 line: 1,
                 column: 13,
                 nodeType: "ExpressionStatement",
-                fix: { range: [12, 12], text: ";" }
+                fix: { range: [12, 12], text: ";" },
+                suggestions: [
```

Comment:
Currently, I don't have any tests for this data getting passed out of the results of the `executeOnFiles` and `executeOnText` methods. It seems like it might be best to test it with an actual rule that implements a suggestion. I feel reasonably confident that I could figure out how to implement the `no-useless-escape` suggestions mentioned [here](https://github.com/eslint/rfcs/pull/30#issuecomment-511543730), and then use that in a test, but I'm not sure if I should include that in this PR. Any suggestions on what to do for those tests?

Reply:
Update: I implemented suggestions for the `no-useless-escape` rule and I do think it is probably valuable to include it in this PR as it will make tests easier, have an actual use case to verify things are set up properly, and will force us to update `RuleTester` so that suggestions can be tested with it. You can see my current work-in-progress commit on that work [here](https://github.com/eslint/eslint/commit/3ac3c9a5bb059e6fc7b2ded21a66cb6b2e512c06). Currently, I'm not thrilled with the way that suggestions can be tested with `RuleTester`, which is the main reason I haven't pushed that commit to this branch.

Reply:
Update 2: I have included the suggestions for the `no-useless-escape` rule in this PR now